macros/contexts/contextBoolean.pl

```diff
@@ -0,0 +1,253 @@
+}
+
+sub typeRef { return Value::Type('Boolean') }
+
+package context::Boolean::BOP;
+our @ISA = qw(Parser::BOP);
+
+sub _check {
+	my $self = shift;
+	# constants don't have type methods
+	# TODO: should that be "fixed"?
+	my $lType = $self->{lop}->type || $self->{lop}{type};
+	my $rType = $self->{rop}->type || $self->{lop}{type};
+
+	$self->{type} = Value::Type('Boolean');
+}
+
+sub perl {
+	my $self   = shift;
+	my $result = $self->SUPER::perl(@_);
+	my $pkg    = $self->Package('Real');
+	return "$pkg->new($result ? 1 : 0)";
+}
+
+package context::Boolean::BOP::or;
+our @ISA = qw(context::Boolean::BOP);
+
+sub _eval { return $_[1] || $_[2] ? 1 : 0 }
+
+package context::Boolean::BOP::and;
+our @ISA = qw(context::Boolean::BOP);
+
+sub _eval { return $_[1] && $_[2] ? 1 : 0 }
+
+package context::Boolean::UOP::not;
+our @ISA = qw(Parser::UOP);
+
+sub _check {
+	my $self  = shift;
+	my $lType = $self->{op}->type || $self->{op}{type};
+
+	$self->{type} = Value::Type('Boolean');
+}
+
+sub _eval { !($_[1]) ? 1 : 0 }
+
+sub perl {
+	my $self   = shift;
+	my $result = $self->SUPER::perl(@_);
+	my $pkg    = $self->Package('Real');
+	return "$pkg->new($result ? 1 : 0)";
+}
+
+package context::Boolean::BOP::xor;
+our @ISA = qw(context::Boolean::BOP);
+
+sub _eval { return !($_[1]) != !($_[2]) ? 1 : 0 }
+
+package context::Boolean::Value;
+our @ISA = qw(Value::Real);
+
+# subclass is necessary because Parser::Constant::getValueType attempts
+# to dereference the incoming value as an ARRAY -- unclear why that is
+sub getValueType {
+	# note value & equation are reversed from Value::getValueType when subclassed
+	my $value    = shift;
+	my $equation = shift;
+	return ($value,                               $value->typeRef) if ref($value) eq 'context::Boolean::Value';
+	return (context::Boolean::Value->new($value), Value::Type('Boolean'));
+}
+
+sub isBoolean {1}
+sub getType   {'Boolean'}
+sub typeRef   { Value::Type('Boolean') }
+sub class     {'Value'}
+sub type      {'Boolean'}
+
+## WIP!
+package context::Boolean::Boolean;
+our @ISA = qw(Parser::Number);
+
+sub new {
+	my $self = shift;
+	warn "NUMBER PARSER";
+	warn "self: $self received " . scalar @_ . " arguments";
+	warn 'arg1 (parser): {' . join(',', keys %{ $_[0] }) . '}';
+	warn "arg2 (token): '$_[1]'";
+	warn "arg3 (tokenRef): $_[2] = ['" . join('\', \'', @{ $_[2] }) . '\']';
+	my $class = ref($self) || $self;
+	my ($equation, $value, $ref) = @_;
+	# $value = $value->value while Value::isReal($value);
+	$value = $value->value while Value::isValue($value);
+	my $num = bless {
+		value        => $value ? 1   : 0,
+		value_string => $value ? 'T' : 'F',
+		type         => Value::Type('Boolean'),
+		isConstant   => 1,
+		ref          => $ref,
+		equation     => $equation,
+	}, $class;
+	$num->weaken;
+	$num->{isOne}  = 1 if $value;
+	$num->{isZero} = 1 unless $value;
+	return $num;
+}
+
+#sub class { 'Boolean' }
+sub typeRef { Value::Type('Boolean') }
+sub getType {'Boolean'}
+sub type    {'Boolean'}
+
```

Comment:
There is no need for this package.  The regular `Parser::Number` should handle the booleans just fine as is.  Again, you should not be modifying type, as it should remain `Number`, as it is for reals, complexes, fractions, etc.

If you really want to `Boolean` type that is distinct from `Number`, you could do that, but I doubt it is worth it, and I'm not sure what the advantage is, as it will really make no difference within the Parser, and the Value object already has class equal to `Boolean`, so I'm not sure what the gain is.